tests/functional/heketi/test_heketi_device_operations.py

```diff
@@ -696,3 +725,92 @@ def test_device_settags_tier_option(self):
+        except (KeyError, IndexError):
+            self.skipTest(
+                "Additional disk is not specified for node with following "
+                "hostnames and IP addresses: {}, {}".format(
+                    gluster_server_0.get('manage', '?'),
+                    gluster_server_0.get('storage', '?')))
+
+        # Get existing heketi volume list
+        existing_volumes = heketi_volume_list(h_client, h_server, json=True)
+
+        # Add cleanup function to clean stale volumes created during test
+        self.addCleanup(
+            self._cleanup_heketi_volumes, existing_volumes.get("volumes"))
+
+        # Get nodes info
+        node_id_list = heketi_node_list(h_client, h_server)
+        node_info_list = []
+        for node_id in node_id_list[:3]:
+            node_info = heketi_node_info(
+                h_client, h_server, node_id, json=True)
+            node_info_list.append(node_info)
+
+        # Disable 4th and other nodes
+        if len(node_id_list) > 3:
+            for node_id in node_id_list[3:]:
+                heketi_node_disable(h_client, h_server, node_id)
+                self.addCleanup(
+                    heketi_node_enable, h_client, h_server, node_id)
+
+        # Create volume when 3 nodes are online
+        vol_size, vol_count = 2, 4
+        for _ in range(vol_count):
+            vol_info = heketi_blockvolume_create(
+                h_client, h_server, vol_size, json=True)
+            self.addCleanup(
+                heketi_blockvolume_delete, h_client, h_server, vol_info['id'])
+
+        # Get node ID of the Gluster hostname
+        topology_info = heketi_topology_info(
+            h_client, h_server, json=True)
+        self.assertIsNotNone(topology_info, "Failed to get topology info")
+        self.assertIn(
+            "clusters", topology_info.keys(), "Failed to get cluster "
+            "details from topology info")
+        node_list = topology_info["clusters"][0]["nodes"]
+        self.assertTrue(
+            node_list, "Cluster info command returned empty list of nodes")
+
+        node_id = None
+        for node in node_list:
+            if manage_hostname == node['hostnames']["manage"][0]:
+                node_id = node["id"]
+                break
+        self.assertTrue(
+            node_id, "No information about node id for {}".format(
+                manage_hostname))
+
+        # Add extra device, then remember it's ID and size
+        device_id_new, _ = self._add_heketi_device(
+            add_device_name, node_id)
+
+        # Remove one of the existing devices on node except new device
+        device_details = []
+        node_info_after_addition = heketi_node_info(
+            h_client, h_server, node_id, json=True)
+        for device in node_info_after_addition["devices"]:
+            if device["name"] != add_device_name:
+                device_details.append((device["name"], device["id"]))
```

Comment:
We are using only one device from this list next steps. Iterating even after meeting this codition will add unnecessary cycles. Break loop here and instead of using list to add this device, we can use `device_name` and `device_id`

Reply:
Done